server/server.ts

```diff
@@ -1,145 +1,160 @@
+  queryString = queryString.replace(
+    /\b(gte|gt|lte|lt|eq|ne|regex|options|in|nin)\b/g,
+    (match) => `$${match}`
+  );
 
-  res.send(exercises)
+  const exercises = await Exercise.find(JSON.parse(queryString));
 
-})
+  res.send(exercises);
+});
 
 app.delete("/api/exercise/:id", async (req, res) => {
-  const result = await Exercise.deleteOne({ _id: req.params.id })
+  const result = await Exercise.deleteOne({ _id: req.params.id });
   if (result) {
-    res.send(result)
+    res.send(result);
   } else {
-    res.send({ "result": "No Record Found" })
+    res.send({ result: "No Record Found" });
   }
-})
+});
 
 app.get("/api/exercise/:id", async (req, res) => {
-  const result = await Exercise.findOne({ _id: req.params.id })
+  const result = await Exercise.findOne({ _id: req.params.id });
   if (result) {
-    res.send(result)
+    res.send(result);
   } else {
-    res.send({ "result": "No Record Found" })
+    res.send({ result: "No Record Found" });
   }
-})
+});
 
 app.get("/api/exercise/:id/relatedExercises", async (req, res) => {
-  const exerciseToGetRealted = await Exercise.findOne({ _id: req.params.id }).exec();
+  const exerciseToGetRealted = await Exercise.findOne({
+    _id: req.params.id,
+  }).exec();
   if (!exerciseToGetRealted) {
-    res.send({ "result": "No Record Found" })
-    return
+    res.send({ result: "No Record Found" });
+    return;
   }
 
-  if (!exerciseToGetRealted.related_to || exerciseToGetRealted.related_to.length === 0) {
-    res.send([])
+  if (
+    !exerciseToGetRealted.related_to ||
+    exerciseToGetRealted.related_to.length === 0
+  ) {
+    res.send([]);
     return;
   }
 
-  const result = await Exercise.find({ "$or": exerciseToGetRealted.related_to.map(r => ({ _id: r._id })) })
+  const result = await Exercise.find({
+    $or: exerciseToGetRealted.related_to.map((r) => ({ _id: r._id })),
+  });
   if (result) {
-    res.send(result)
+    res.send(result);
   } else {
-    res.send({ "result": "No Record Found" })
+    res.send({ result: "No Record Found" });
   }
-})
+});
 
 app.put("/api/exercise/:id", async (req, res) => {
   const result = await Exercise.updateOne(
     { _id: req.params.id },
-    { $set: req.body })
-  res.send(result)
-})
+    { $set: req.body }
+  );
+  res.send(result);
+});
 
 app.get("/api/search/:key", async (req, res) => {
   const result = await Exercise.find({
-    "$or": [
+    $or: [
       {
         name: { $regex: req.params.key },
-
-      }
+      },
       // ,
       // {
       //   // name: { $regex: req.params.key },
       //   tags: { $regex: req.params.key },
       // }
-    ]
-  })
+    ],
+  });
   if (result) {
-    res.send(result)
+    res.send(result);
   } else {
-    res.send({ "result": "No Record Found" })
+    res.send({ result: "No Record Found" });
   }
-})
+});
+
+app.get("/api/tags", async (req, res) => {
+  // gets all distinct values of tags
+  const result = await Exercise.distinct("tags");
```

Comment:
Check here how to filter the Execise.distinct("tags") by passed query string using regex like inside app.get("/api/exercises", async (req, res) => {